Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,796 @@
+//
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Text;
+#if !NO_SYSTEM_WEB_DEPENDENCY && !MOBILE && false
+using System.Web.Configuration;
+#endif
+
+namespace System.Web.Util
+{
+    public
+    class HttpEncoder
+    {
+        static char [] hexChars = "0123456789abcdef".ToCharArray ();
+        static object entitiesLock = new object ();
+        static SortedDictionary <string, char> entities;
+        static Lazy <HttpEncoder> defaultEncoder;
+        static Lazy <HttpEncoder> currentEncoderLazy;
+        static HttpEncoder currentEncoder;
+
+        static IDictionary <string, char> Entities {
+            get {
+                lock (entitiesLock) {
+                    if (entities == null)
+                        InitEntities ();
+
+                    return entities;
+                }
+            }
+        }
+        
+        public static HttpEncoder Current {
+            get {
+                if (currentEncoder == null)
+                    currentEncoder = currentEncoderLazy.Value;
+                return currentEncoder;
+            }
+            set {
+                if (value == null)
+                    throw new ArgumentNullException ("value");
+                currentEncoder = value;
+            }
+        }
+
+        public static HttpEncoder Default {
+            get {
+                return defaultEncoder.Value;
+            }
+        }
+
+        static HttpEncoder ()
+        {
+            defaultEncoder = new Lazy <HttpEncoder> (() => new HttpEncoder ());
+            currentEncoderLazy = new Lazy <HttpEncoder> (new Func <HttpEncoder> (GetCustomEncoderFromConfig));
+        }
+        
+        public HttpEncoder ()
+        {
+        }
+        protected internal virtual
+        void HeaderNameValueEncode (string headerName, string headerValue, out string encodedHeaderName, out string encodedHeaderValue)
+        {
+            if (String.IsNullOrEmpty (headerName))
+                encodedHeaderName = headerName;
+            else
+                encodedHeaderName = EncodeHeaderString (headerName);
+
+            if (String.IsNullOrEmpty (headerValue))
+                encodedHeaderValue = headerValue;
+            else
+                encodedHeaderValue = EncodeHeaderString (headerValue);
+        }
+
+        static void StringBuilderAppend (string s, ref StringBuilder sb)
+        {
+            if (sb == null)
+                sb = new StringBuilder (s);
+            else
+                sb.Append (s);
+        }
+        
+        static string EncodeHeaderString (string input)
+        {
+            StringBuilder sb = null;
+            
+            for (int i = 0; i < input.Length; i++) {
+                char ch = input [i];
+
+                if ((ch < 32 && ch != 9) || ch == 127)
+                    StringBuilderAppend (String.Format ("%{0:x2}", (int)ch), ref sb);
+            }
+
+            if (sb != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)